sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -361,8 +362,8 @@ struct SocRegistersImpl {
     #[register(offset = 0x00b0)]
     cptra_timer_config: ReadWriteRegister<u32>,
 
-    #[register(offset = 0x00b4)]
-    cptra_bootfsm_go: ReadOnlyRegister<u32>,
+    #[register(offset = 0x00b4, write_fn = on_write_bootfsm_go)]
+    cptra_bootfsm_go: u32,
```

Comment:
Why are we using a primitive type u32 instead of making a ReadWriteRegister<u32>?

Reply:
ReadWriteRegister doesn't really do anything useful over a primitive. ReadOnlyRegister is handy since it lets you prevent writes without having to write any code. In this case, we have a write_fn that handles the write behavior, and the default read behavior of "return the primitive" is what we want.